code/__HELPERS/maths.dm

```diff
@@ -30,34 +30,44 @@
  * Uses the ultra-fast [Bresenham Line-Drawing Algorithm](https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm).
  */
 /proc/get_line(atom/starting_atom, atom/ending_atom)
-	var/px = starting_atom.x //starting x
-	var/py = starting_atom.y
-	var/line[] = list(locate(px, py, starting_atom.z))
-	var/dx = ending_atom.x - px //x distance
-	var/dy = ending_atom.y - py
-	var/dxabs = abs(dx)//Absolute value of x distance
-	var/dyabs = abs(dy)
-	var/sdx = SIGN(dx) //Sign of x distance (+ or -)
-	var/sdy = SIGN(dy)
-	var/x = dxabs >> 1 //Counters for steps taken, setting to distance/2
-	var/y = dyabs >> 1 //Bit-shifting makes me l33t.  It also makes get_line() unnessecarrily fast.
-	var/j //Generic integer for counting
-	if(dxabs >= dyabs) //x distance is greater than y
-		for(j = 0; j < dxabs; j++)//It'll take dxabs steps to get there
-			y += dyabs
-			if(y >= dxabs) //Every dyabs steps, step once in y direction
-				y -= dxabs
-				py += sdy
-			px += sdx //Step on in x direction
-			line += locate(px, py, starting_atom.z)//Add the turf to the list
+	var/current_x_step = starting_atom.x//start at x and y, then add 1 or -1 to these to get every turf from starting_atom to ending_atom
+	var/current_y_step = starting_atom.y
+	var/starting_z = starting_atom.z
+
+	var/list/line[] = list(get_turf(starting_atom))//get_turf(atom) is faster than locate(x, y, z)
+
+	var/x_distance = ending_atom.x - current_x_step //x distance
+	var/y_distance = ending_atom.y - current_y_step
+
+	var/abs_x_distance = abs(x_distance)//Absolute value of x distance
+	var/abs_y_distance = abs(y_distance)
+
+	var/x_distance_sign = SIGN(x_distance) //Sign of x distance (+ or -)
+	var/y_distance_sign = SIGN(y_distance)
+
+	var/x = abs_x_distance >> 1 //Counters for steps taken, setting to distance/2
+	var/y = abs_y_distance >> 1 //Bit-shifting makes me l33t.  It also makes get_line() unnessecarrily fast.
```

Comment:
(it didnt make it faster according to tests but i didnt want to steal his thunder)

Reply:
(also two divisions isnt going to be a major component of this procs cost anyways)

Reply:
lol